File: api/src/classes/units/Command.js
const Infantry = require('./infantry')

class Command extends Infantry {
  constructor(square, team) {
    super(square, team)
    this.type = "command"
  }

  afterMove() {
    super()
    this.square
      .neighbors({headings: [[0, 1]], units: true})
      .filter(unit => unit.team == this.team)
      .forEach(unit => unit.defense_modifier++)
  }
}

module.exports = Command

File: api/lib/units/command.rb
require_relative './infantry.rb'

class Command < Infantry
  def initialize square, team
    super
    @type = :command
  end

  # +1 defense to orthogonally adjacent friendlies
  def after_move
    super
    @square
      .neighbors(headings:[[0, 1]], units: true)
      .each { |unit| unit.defense_modifier++ if unit.team == @team }
  end
end